models/gating_network.py

```diff
@@ -0,0 +1,28 @@
+from turtle import forward
+import torch.nn as nn
+import torch
+
+
+class Gating_Network(nn.Module):
+    def __init__(self, activation, args):
+        super().__init__()
+        m = nn.Conv2d(activation.shape[1], 30, 1, stride=2).cuda()
+        k = m(activation)
+        self.neurons_fc = k.shape[1] * k.shape[2] * k.shape[3]
+        self.first_conv = nn.Conv2d(activation.shape[1], 30, 1, stride=2).cuda()

```

Comment:
Alternatively it is quite common to do global average (or max) pooling after a convolution that has tens (or even more) of output channels. Then as a bonus you don't have to do any inference (or calculation) of output size in the constructor, and it works for any input image size. On the other hand this is a different architecture that is not guaranteed to work better than this one, so I would leave it as it is for now.